### ISSUE TITLE
[Masthead A11Y]: Multiple elements with "navigation" role do not have unique labels

### ISSUE THREAD

**Issue by mo-alaa** [carbon-design-system/carbon-for-ibm-dotcom#9680]
### Description

Hi, while using the L1 masthead (contained in DotcomShell @ ibmdotcom-react), we can find that the "navigation" elements do not have unique labels.

This was raised as a violation while scanning [the page](https://www.ibm.com/impact) (L1 is used in that page) with [IBM Equal Access Accessibility Checker](https://chrome.google.com/webstore/detail/ibm-equal-access-accessib/lkcagbfjnkomcinoddgooolagloogehp/related), the aria-label doesn't have a label.

I believe this also occurs on the masthead of _ibmdotcom-web-components_.
Thanks!
![image](https://user-images.githubusercontent.com/41386480/203323062-a6fe5f57-a715-432f-9a3a-655afd81666b.png)

### Component(s) impacted

DotcomShell of ibmdotcom-react

### Browser

Chrome

### Carbon for IBM.com version

Carbon for IBM.com React v1.40.0

### Severity

Severity 3 = The problem is visible or noticeable to users but does not impede the usability or functionality. Affects minor functionality, has a workaround.

### Application/website

ibm.com/impact

### Package

@carbon/ibmdotcom-react, @carbon/ibmdotcom-web-components

### CodeSandbox example

https://www.ibm.com/standards/carbon/react/?path=/story/components-masthead--with-l-1

### Steps to reproduce the issue (if applicable)

his can be inspected in the [docs for the masthead L1 element](https://www.ibm.com/standards/carbon/react/?path=/story/components-masthead--with-platform) as shown in the images:

- [ibmdotcom-react](https://www.ibm.com/standards/carbon/react/?path=/story/components-masthead--with-platform):
![image](https://user-images.githubusercontent.com/41386480/203324248-622936d5-6839-49e1-8c57-b1013fd33aae.png)

<hr/>

- I believe it occurs in[ ibmdotcom-web-components](https://www.ibm.com/standards/carbon/web-components/react/?path=/story/components-masthead--with-l-1) as well:
![image](https://user-images.githubusercontent.com/41386480/203324360-dd4f3660-af6c-4f32-9e8a-96cb41660b7d.png)

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/main/.github/CODE_OF_CONDUCT.md)
- [ ] I checked the [current issues](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues) for duplicate issues

**Comment by mo-alaa** [comment]
Just a heads-up. Thanks!
